src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -66,39 +78,81 @@ const TopicFragment = ({
     parseToken();
   }, []);
 
-  const handleFollow = async (from, willFollow, item, index) => {
+  const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+    const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+    newTopicLists[indexTopic].following = !!willFollow;
+    newTopicLists[indexTopic].user_id_follower = myId;
+    return newTopicLists[indexTopic];
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `exhangeFollower` is mutating the `newTopicLists` array directly which is an anti-pattern in React. It's better to create a new copy of the array before making changes to avoid unexpected side effects.
```diff
-   const exhangeFollower = (newTopicLists, willFollow, topicId) => {
-     const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
-     newTopicLists[indexTopic].following = !!willFollow;
-     newTopicLists[indexTopic].user_id_follower = myId;
-     return newTopicLists[indexTopic];
-   };
+   const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+     const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+     const updatedTopics = [...newTopicLists];
+     updatedTopics[indexTopic] = {
+       ...updatedTopics[indexTopic],
+       following: !!willFollow,
+       user_id_follower: myId
+     };
+     return updatedTopics[indexTopic];
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->